cluster/kubernetes/images.go

```diff
@@ -14,7 +15,18 @@ import (
 	"github.com/weaveworks/flux/registry"
 )
 
-func mergeCredentials(log func(...interface{}) error, client extendedClient, namespace string, podTemplate apiv1.PodTemplateSpec, imageCreds registry.ImageCreds, seenCreds map[string]registry.Credentials) {
+func mergeCredentials(log func(...interface{}) error,
+	filterImages func(podTemplate apiv1.PodTemplateSpec) []image.Name,
+	client extendedClient,
+	namespace string, podTemplate apiv1.PodTemplateSpec,
+	imageCreds registry.ImageCreds,
+	seenCreds map[string]registry.Credentials) {
+	// filter the images based on the exclusion list
+	images := filterImages(podTemplate)
```

Comment:
All that happens to `filterImages` is that it's applied to one of the other arguments -- you may as well supply the result.